File: Leetcode/LC75/ContainsDuplicate2/Solution.go
func containsNearbyDuplicate(nums []int, k int) bool {
    m := make(map[int]int)
    
    for idx, val := range nums {
        if val, ok := m[val]; ok {
            if idx - val <= k {
                return true
            }
        }
        m[val] = idx
    }
    
    return false
}

File: Leetcode/LC75/ContainsDuplicate2/Solution.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int n = nums.length;
        HashMap<Integer, Integer> m = new HashMap<>();
        
        for(int i=0; i<n; i++) {
            if(m.containsKey(nums[i])) {
                if(i - m.get(nums[i]) <= k) return true;
            }
            
            m.put(nums[i], i);
        }
        return false;
    }
}
